src/Elf.h

```diff
@@ -283,7 +290,15 @@ class Section {
         this->contents.insert(this->contents.end(), begin, end);
         return *this;
     }
-    /** Set or append an object to the contents, assuming T is a
+    template <typename It>
+    Section &prepend_contents(It begin, It end, uint64_t off) {
```

Comment:
`off` -> `size_bytes`? That said, it's weird that this function takes both an iterator range, and what is presumably the size of the data being prepended (technically redundant, redundancies like this are opportunities for bugs). What about removing the argument, and doing:

    uint64_t size_bytes = std::distance(begin, end) * sizeof(T)?

Another possibility is that I'm torn about the presence of prepend_contents at all (due to modifying the relocations). I wonder if this should be implemented by the caller of this just getting the contents of the section, prepending, and then setting the contents after.